neps/nep-0483.md

```diff
@@ -0,0 +1,83 @@
+---
+NEP: 0461
+Title: Prefix Tag for Signed Messages
+Author: Evgeny Kapun <eatmore@near.org>, Gallardo Guillermo <guillermo@near.org>, Firat Sertgoz <firat@near.org>
+DiscussionsTo: https://github.com/nearprotocol/neps/pull/461
+Status: Review
+Type: Protocol
+Category: Runtime
+Version: 1.0.0
+Created: 14-Feb-2023
+Updated: 14-Feb-2023
+---
+
+## Summary
+
+A proposal to include a standardized prefix in all standardized signable messages, to distinguish between (1) actionable messages (i.e. those that are validated on chain), (2) non-actionable messages (i.e. those that are used off-chain), and (3) [classic transactions](https://nomicon.io/RuntimeSpec/Transactions).
+
+Particularly, we propose to prepend an explicit prefix to the structure being signed using the following rules:
+
+| Prefix | Message |
+| - | - |
+| $< 2^{30}$ | [Classic Transactions](https://nomicon.io/RuntimeSpec/Transactions) |  
+| $2^{30}$ + NEP-number  | On-chain Actionable Message, e.g. [`SignedDelegateAction`](https://github.com/near/NEPs/blob/master/neps/nep-0366.md) |
+| $2^{31}$ + NEP-number  | Off-chain Message, e.g. [`signMessage`](https://github.com/near/NEPs/pull/413) |
```

Comment:
`original comment by` @mfornet

Should we explicitly reserve the range [3 * 2^30, 4 * 2^30) as unused / experimentation / undefined?